src/core/weightedGraph2.js

```diff
@@ -37,89 +65,252 @@ export type WeightedEdge = {|
  * WeightedGraph2 should completely replace it.
  */
 export class WeightedGraph2 {
```

Comment:
From our first refactoring call, I get the sense that composability will matter a lot with these types.  I find that it's easier / more concise to compose objects than classes (in that classes are just objects with more complications), but that's just me.  If there's a benefit of using a class here, fine by me.  If not, I lean towards the style used in `Weights` and `Node/EdgeWeights`.

Reply:
@decentralion would be curious to get your opinion on this.

I was trying to follow the structure of the `Graph` class moreso than `WeightsI`, unless `Graph` is scheduled to be refactored to follow `WeightsI`'s object-literal return instead of an ES6 class